### ISSUE TITLE
Automoderator takes actions on shadowbanned users even though "exclude posts by site-wide banned users from modqueue/unmoderated" has been enabled. 

### ISSUE THREAD

**Issue by creesch** [reddit/reddit#1497]
@Deimos 

Pretty much as the title says. It seems like automoderator does operate based on its own queue that ignores this setting. 

An example would be in /r/gadgets where we operate on a whitelist basis and ask people submitting something not on the list to message us to get it approved. This message is also triggered for shadowbanned users which in /r/gadgets are almost always spammers.

**Comment by jkingsman** [comment]
This is an issue with automoderator, not a bug with the reddit platform, no?

**Comment by Pokechu22** [comment]
Automoderator is actually [built-in to the site](https://github.com/reddit/reddit/blob/master/r2/r2/lib/automoderator.py) now (although previously it was a separate bot).  This happened [about 9 months ago](https://www.reddit.com/r/changelog/comments/30z70k/reddit_change_automoderator_is_now_integrated/).

**Comment by jkingsman** [comment]
I stand corrected :) Nevermind then!

**Comment by Jakeable** [comment]
As a temporary fix, you can add this to your conditions to prevent it from triggering on shadowbanned users:

```
author:
        link_karma: "> -1001"
```

**Comment by 13steinj** [comment]
@Jakeable that's a strange fix if I ever saw one. Do shadowbanned users have specifically less than -1000 karma in automod's eyes?

**Comment by jkingsman** [comment]
And wouldn't it need to be `< -1001`? 

`"> -1001"` will match for greater than or equal to -1000 (the majority of users).

**Comment by Deimos** [comment]
Hmm, I think I'd consider this more of an "expectations mismatch" than a "bug". The "exclude posts by site-wide banned users from modqueue/unmoderated" subreddit setting just makes it so that posts by shadowbanned users are hidden from those two listing pages, but AutoModerator doesn't "use" those pages at all (though it used to when it was still an external bot), it just processes every post made to the subreddit. I'm not really sure if tying that setting to AutoModerator behavior makes much sense, or maybe if there should be some specific way of telling AutoMod to skip over posts by shadowbanned users.

As for @Jakeable's fix, the reason it works is that shadowbanned users always "fail" any checks on `author:` attributes. It doesn't have to be a specific attribute/number or anything, something like this should work as well:

    author:
        account_age: '> 0 minutes'

**Comment by creesch** [comment]
Fair enough,  a setting would be appreciated. Meanwhile I'll use the check on author as a workaround :)
